### PR TITLE
CTBase 0.13

### PR DESCRIPTION
**Pull request by PierreMartinon**
Solution constructors updated.
Manual check on multipliers / constraints looks fine.

Todo: update the JSON export/read (probably fuse into save/load with a format argument)

Edit: 
- the missing JSON part is probably why the docs build fails
- OC tests fail too, I'll check this

### CODE REVIEW

**Comment by PierreMartinon** [comment]
> Package name 	latest 	stable
> OptimalControl.jl 	[](https://github.com/control-toolbox/CTDirect.jl/actions/runs/10528667868) 	[](https://github.com/control-toolbox/CTDirect.jl/actions/runs/10528667868)

@ocots the badge correctly indicates a failure while the breakage actions themselves display a success depsite the errors. Can we fix that or is it required by the integration somehow ?

**Comment by ocots** [comment]
There is a failure since OC need CTBase.jl 0.12. It is normal.

The check can be ok even if there is a failure since the check is for merging. The failure indicates that there is something to change in OC.

File: ext/CTDirectExt.jl [hunk 1]
```diff
@@ -32,9 +32,10 @@ $(TYPEDSIGNATURES)
   
 Export OCP solution in JSON format
 """
+# +++ redo this, start with basics
```
Comment (ocots) [inline ext/CTDirectExt.jl:35]: Attention les commentaires entre la documentation et la fonction créent une erreur dans la génération de la documentation : 

https://github.com/control-toolbox/CTDirect.jl/actions/runs/10528667872/job/29174707511?pr=216#step:6:12

File: ext/CTDirectExt.jl [hunk 2]
```diff
@@ -44,9 +45,10 @@ $(TYPEDSIGNATURES)
   
 Read OCP solution in JSON format
 """
+# +++ add constructor from json blob
```
Comment (ocots) [inline ext/CTDirectExt.jl:48]: Idem ici.

**Comment by ocots** [comment]
@PierreMartinon Avant de merge cette PR, il faudrait merge celle-ci : https://github.com/control-toolbox/CTDirect.jl/pull/218

Attention, https://github.com/control-toolbox/CTDirect.jl/pull/218 merge dans la branche `ctbase0.13`. C'est juste un formattage du code. Je suis en train d'ajouter un formattage automatique sur tous les repos.

**Comment by ocots** [comment]
@PierreMartinon Tu veux pas la merge ?

**Comment by PierreMartinon** [comment]
> @PierreMartinon Tu veux pas la merge ?

La 218 est faite, je suis en train de finir celle-ci

**Pull request merged**